### PR TITLE
Add robots.txt and sitemap.xml

### PR DESCRIPTION
**Pull request by lpsinger**
See https://github.com/remix-run/remix/discussions/2912#discussioncomment-7039395

## Checklist

- [ ] Tests updated
- [ ] Docs updated

### CODE REVIEW

**Comment by lpsinger** [comment]
I'm not sure what's going wrong when I run `npm run dev`:

```
$ npm run dev

> predev
> npm run build:icons --silent


> dev
> remix dev -c "node ./server/dev-server.js" --manual


 💿  remix dev

 info  building...
 info  built (8.4s)
6:30:10 PM [tsx] rerunning
Debugger listening on ws://127.0.0.1:9229/8911dd2a-b040-4bac-ac18-7207cb0b3a3c
For help, see: https://nodejs.org/en/docs/inspector
TypeError: Cannot read properties of undefined (reading 'startsWith')
TypeError: Cannot read properties of undefined (reading 'startsWith')
    at file:///Users/lpsinger/src/epic-stack/tests/mocks/github.ts:1:2646
```

File: app/routes/_seo+/sitemap[.]xml.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
```
Comment (kentcdodds) [inline app/routes/_seo+/sitemap[.]xml.ts:2]: I understand why this has to be here, but I don't know how I feel about merging this with the copyright disclaimer.

Do I need to put together a CLA for the first time in my life? 😬

**Review by kentcdodds — COMMENTED** [review]
Thanks for this! I can fix the issue you're having. But I'm concerned about something else.

Also, we probably should add some documentation.

File: app/routes/_seo+/sitemap[.]xml.ts (hunk 1 above)
Comment (lpsinger) [reply app/routes/_seo+/sitemap[.]xml.ts:2]: Oops, I forgot to remove that.

**Review by kentcdodds — APPROVED** [review]
Thanks a bunch! This is very useful!

**Pull request merged**